File: 백준/Bronze/9506. 약수들의 합/약수들의 합.py
n = int(input())
while n != -1:
    aliquot = []
    for i in range(1, n):
        if n % i == 0:
            aliquot.append(i)
    if sum(aliquot) == n:
        print(str(sum(aliquot)) + " = ", end="")
        for i in range(len(aliquot) - 1):
            print(str(aliquot[i]) + " + ", end="")
        print(str(aliquot[len(aliquot) - 1]))
    else:
        print(f"{n} is NOT perfect.")

    n = int(input())

File: 백준/Bronze/9506. 약수들의 합/약수들의 합.java
import java.io.*;
import java.util.*;

public class Main {
	public static void main(String[] args) throws IOException {
		Scanner sc = new Scanner(System.in);

		int n = sc.nextInt(); // 주어진 수 입력받기
		while (n != -1) {
			StringBuilder sb = new StringBuilder();
			int sum = 0;

			for (int i = 1; i < n; i++) {
				if (n % i == 0) {
					sb.append(i).append(" + ");
					sum += i;
				}
			}

			if (sum == n) {
				System.out.println(n + " = " + sb.toString().substring(0, sb.length() - 3));
			} else {
				System.out.println(n + " is NOT perfect.");
			}
			n = sc.nextInt();
		}

		sc.close();
	}
}